### PR TITLE
chore: remove python 3.8 references

### PR DESCRIPTION
**Pull request by Danyal-Faheem**
Achieves part of [#1129](https://github.com/overhangio/tutor/issues/1129).

- Update python 3.8 references to 3.9
- Use Github actions to use 3.9
- Ensure tests are working as expected on 3.9

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def load_about():
         "License :: OSI Approved :: GNU Affero General Public License v3",
         "Operating System :: OS Independent",
         "Programming Language :: Python",
-        "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
```
Comment (DawoudSheraz) [inline setup.py:55]: 3.9 is mentioned below already, you can remove this line.

Comment (Danyal-Faheem) [reply setup.py:55]: Updated. Thanks for pointing it out.

File: changelog.d/20241107_185222_danyal.faheem_remove_py3_8_reference.md [hunk 2]
```diff
@@ -0,0 +1 @@
+- [Depreciation] Drop support for python 3.8 as it has reached end of life. (by @Danyal-Faheem)
```
Comment (DawoudSheraz) [inline changelog.d/20241107_185222_danyal.faheem_remove_py3_8_reference.md]: nit: In changelog template, its `Deprecation`, not `Depreciation`

Comment (Danyal-Faheem) [reply changelog.d/20241107_185222_danyal.faheem_remove_py3_8_reference.md]: Whoops, thanks for picking it out.

**Review by DawoudSheraz — APPROVED** [review]

**Pull request merged**